lib/ui/compositing/scene.cc

```diff
@@ -102,7 +98,41 @@ Dart_Handle Scene::toImage(uint32_t width,
   return Picture::RasterizeToImage(picture, width, height, raw_image_callback);
 }
 
-std::unique_ptr<flutter::LayerTree> Scene::takeLayerTree() {
+void Scene::RasterizeToImageSync(uint32_t width,
+                                 uint32_t height,
+                                 Dart_Handle raw_image_handle) {
+  auto* dart_state = UIDartState::Current();
```

Comment:
I know this is a private method, but we should probably at _least_ `FML_DCHECK(layer_tree_)` here.

Also, you'll need to do a null check on `dart_state` and just bail out early if it's nullptr (I think that might be extra defensive but there ends up being weirdness around isolate teardown and when these objects get destroyed).

Reply:
Done